ast/src/parsed/mod.rs

```diff
@@ -42,6 +42,8 @@ pub enum SymbolCategory {
     TypeConstructor,
     /// A trait declaration, which can be used as a type.
     TraitDeclaration,
+    /// A struct
```

Comment:
If you introduce this and it cannot be unified with `Type` then please also rename `Type`.